### PR TITLE
split the dateTime type into 4 types

### PR DESCRIPTION
**Pull request by FrankYFTang**

### CODE REVIEW

File: spec.emu [hunk 1]
```diff
@@ -102,97 +112,166 @@ contributors: Google, Ecma International
             </tr>
           </thead>
           <tr>
-            <td>`"sunday"`</td>
-            <td>Name indicating Sunday, e.g. Su, Sun, Sunday, etc.</td>
-          </tr>
-          <tr>
-            <td>`"monday"`</td>
+            <td>`"1"`</td>
```
Comment (sffc) [inline spec.emu]: Should be a number, not a string, right?  Or is the argument coerced to a string?

File: spec.emu [hunk 2]
```diff
@@ -102,97 +112,166 @@ contributors: Google, Ecma International
...
             <td>Name indicating October, e.g. Oct, October, etc.</td>
           </tr>
           <tr>
-            <td>`"november"`</td>
+            <td>`"11"`</td>
             <td>Name indicating November, e.g. Nov, November, etc.</td>
           </tr>
           <tr>
-            <td>`"december"`</td>
+            <td>`"12"`</td>
```
Comment (sffc) [inline spec.emu]: Please add month 13 as per #48

File: spec.emu [hunk 3]
```diff
@@ -102,97 +112,166 @@ contributors: Google, Ecma International
...
+        <emu-caption>Codes For Day Period of DisplayNames</emu-caption>
+        <table class="real-table">
+          <thead>
+            <tr>
+              <th>Code</th>
+              <th>Description</th>
+            </tr>
+          </thead>
           <tr>
             <td>`"am"`</td>
```
Comment (sffc) [inline spec.emu:275]: This is fine for now, but please leave open an issue to decide whether to use am/pm or 1/2.

File: spec.emu (hunk 1 above)
Comment (FrankYFTang) [reply spec.emu]: ToString is called in Intl.DisplayNames.prototype.of( code ) first then call into IsValidCodeForDisplayNames. Therefore, all the code is already turned into String when it passing to IsValidCodeForDisplayNames, even if the input code user passed in is a number.

File: spec.emu [hunk 4]
```diff
@@ -102,97 +112,170 @@ contributors: Google, Ecma International
...
+        <table class="real-table">
+          <thead>
+            <tr>
+              <th>Code</th>
+              <th>Description</th>
+            </tr>
+          </thead>
+          <tr>
+            <td>`"1"`</td>
             <td>Name indicating January, e.g. Jan, January, etc.</td>
```
Comment (sffc) [inline spec.emu]: Why don't you say,

> Name indicating the first month of the year, e.g., Jan, January.

Then the definition is not tied to the Gregorian calendar only.

Comment (FrankYFTang) [reply spec.emu]: DONE

**Comment by FrankYFTang** [comment]
PTAL

File: spec.emu [hunk 5]
```diff
@@ -59,47 +59,37 @@ contributors: Google, Ecma International
...
-          1. Return ! IsValidQuarterCode(_code_).
-        1. If _type_ is `"dayPeriod"`
-          1. Return ! IsValidDayPeriodCode(_code_).
+          1. If _code_ matches the `unicode_script_subtag` production, return *true*.
+          1. Return *false*.
+        1. If _type_ is `"currency"`, return ! IsWellFormedCurrencyCode(_code_).
+        1. If _type_ is `"weekday"`, return ! IsValidWeekdayCode(_code_).
+        1. If _type_ is `"month"`, return ! IsValidMonthCode(_code_).
+        1. If _type_ is `"quarter"`,  return ! IsValidQuarterCode(_code_).
+        1. If _type_ is `"dayPeriod"`, return ! IsValidDayPeriodCode(_code_).
```
Comment (sffc) [inline spec.emu:78]: Why the `!` ... where did you get that syntax from?

File: spec.emu [hunk 6]
```diff
@@ -59,47 +59,37 @@ contributors: Google, Ecma International
...
+      <h1>IsValidWeekdayCode ( _weekday_ )</h1>
+      <p>
+        The IsValidWeekdayCode abstract operation is called with arguments _weekday_. It verifies that the _weekday_ argument represents a valid weekday code. The following steps are taken:
+      </p>
       <emu-alg>
-        1. If _code_ is listed in <emu-xref href="#table-validcodeforweekday"></emu-xref>, then
-          1. Return *true*.
-        1. Else,
-          1. Return *false*.
+        1. Let _weekday_ be ? ToNumber(_weekday_).
```
Comment (sffc) [inline spec.emu:90]: Hmm.  This means things like `true`, `"  1"`, `"1.0"`, etc., all resolve to Monday.  Probably fine, but maybe we should be more strict and keep the string representation?  A question for the editors. @spectranaut @leobalter @ljharb

File: spec.emu [hunk 7]
```diff
@@ -163,69 +155,71 @@ contributors: Google, Ecma International
...
-            <td>`"12"`</td>
-            <td>Name indicating December, e.g. Dec, December, etc.</td>
+            <td>`12`</td>
+            <td>The twelfth month of the year, e.g. Dec, December.</td>
           </tr>
           <tr>
-            <td>`"13"`</td>
-            <td>Name indicating the thirteen month in some non-Gregorian calendar.</td>
+            <td>`13`</td>
+            <td>The thirteenth month, in some non-Gregorian calendar.</td>
```
Comment (sffc) [inline spec.emu]: > The thirteenth month, for relevant calendar systems.

File: spec.emu (hunk 5 above)
Comment (ljharb) [reply spec.emu:78]: That’s how abstract operations assert they don’t return an abrupt completion in 262; all abstract operation calls in 262 that don’t inspect the completion record value should have a ! or ? prefix.

(Not sure if 402 follows the same convention or not)

File: spec.emu (hunk 6 above)
Comment (ljharb) [reply spec.emu:90]: Personally I’d prefer an api that throws on the wrong type, but 262 tends to coerce :-/

File: spec.emu (hunk 7 above)
Comment (FrankYFTang) [reply spec.emu]: DONE

File: spec.emu (hunk 3 above)
Comment (FrankYFTang) [reply spec.emu:275]: > This is fine for now, but please leave open an issue to decide whether to use am/pm or 1/2.

will do so once land this PR. Does not make sense for such issue w/o this PR in the spec.

**Review by sffc — APPROVED** [review]

**Comment by sffc** [comment]
@littledan You expressed some questions/concerns about this change in the meeting last week.  Please take a look so we can revisit if necessary.

**Comment by littledan** [comment]
After thinking about this some more, I don't have any concerns anymore. LGTM.

**Pull request merged**